Fix VectorLoad

lukamac opened the pull request:
The commit f18aba6 is the important one. It fixes the `VectorLoad` to handle correctly the case were the requested load is smaller then `alignment_in_bytes_` and after talking with Francesco this should also kinda model better the hardware.

arpansur commented:
other than that it looks good to me. Thank you for the refactoring!

lukamac commented:
> other than that it looks good to me. Thank you for the refactoring!

other than what?

arpansur commented:
> > other than that it looks good to me. Thank you for the refactoring!
> 
> other than what?

minor correction remaining_size not remainding size :) same for other places it is used the variable name.

lukamac commented:
> > > other than that it looks good to me. Thank you for the refactoring!
> > 
> > 
> > other than what?
> 
> minor correction remaining_size not remainding size :) same for other places it is used the variable name.

Fixed here 195f85d

Additionaly, I made a few other changes:
- added IoMasterAddrMasked which masks the addresses so that address masking is done in one place 5b85d85
- changed infeat loading logic regarding padding bc149b5
